src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -484,7 +484,7 @@ namespace cryptonote
           crypto::generate_ring_signature(tx_prefix_hash, boost::get<txin_to_key>(tx.vin[i]).k_image, keys_ptrs, in_contexts[i].in_ephemeral.sec, src_entr.real_output, sigs.data());
         ss_ring_s << "signatures:" << ENDL;
         std::for_each(sigs.begin(), sigs.end(), [&](const crypto::signature& s){ss_ring_s << s << ENDL;});
-        ss_ring_s << "prefix_hash:" << tx_prefix_hash << ENDL << "in_ephemeral_key: " << in_contexts[i].in_ephemeral.sec << ENDL << "real_output: " << src_entr.real_output << ENDL;
+        ss_ring_s << "prefix_hash:" << tx_prefix_hash << ENDL << "in_ephemeral_key: " << rct::sk2rct(in_contexts[i].in_ephemeral.sec) << ENDL << "real_output: " << src_entr.real_output << ENDL;
```

Comment:
There are unwrap functions to expose the inner type without copy - unwrap(unwrap(txkey.sec)). This occurs several times throughout this PR.

Reply:
The function `rct::sk2rct` also doesn't copy, but `unwrap` is a definitely a lot cleaner

Reply:
The function does a return by value - so it is definitely triggering the copy constructor in this situation.

Reply:
Nevermind I missed the `&` by the function name. Yikes! This violates the aliasing rule, too bad I missed this.

Reply:
Okay newest force push drops usage of `rct::sk2rct` to convert secret keys to `rct::key` and uses `unwrap` or a wrapper called `crypto::secret_key_explicit_print_ref`. I can look into making a PR to fix the strict aliasing later